app/assets/javascripts/application.js.erb

```diff
@@ -205,35 +205,50 @@ var _Lobsters = Class.extend({
       action, { reason: reason });
   },
 
-  postComment: function(form) {
+  postComment: function(btn, form) {
     var params = $(form).serializeArray();
     params.push({"name": "show_tree_lines", "value": "true"});
 
-    $.post($(form).attr("action"), params, function(data) {
-      // Clear form: Firefox will keep form values on reload (e.g. F5), which isn't too useful if
-      // it's already posted.
-      $(form).find('textarea').val('')
-
-      var parsedHTML = $.parseHTML(data);
-
-      var isEdit = $(form).hasClass("edit_comment");
-      var hasChildren = $(form).closest(".comments_subtree")
-        .find(".comments_subtree").length;
-      if (isEdit && hasChildren) {
-        $(parsedHTML).find(".comment_parent_tree_line:first")
-          .removeClass("no_children");
-      }
+    var done = Lobsters.loading(btn, () => {
+      $.post($(form).attr("action"), params, function(data) {
+        // Clear form: Firefox will keep form values on reload (e.g. F5), which isn't too useful if
+        // it's already posted.
+        $(form).find('textarea').val('')
 
-      if ($(form).find("#parent_comment_short_id").length) {
-        // reply to comment
-        $(form).closest(".comments_subtree")
-          .find(".comment_parent_tree_line:first").removeClass("no_children");
-        $(form).closest(".comment").replaceWith(parsedHTML);
-      } else {
-        // reply to story
-        $(form).parent(".comment").replaceWith(parsedHTML);
-      }
-    });
+        var parsedHTML = $.parseHTML(data);
+
+        var isEdit = $(form).hasClass("edit_comment");
+        var hasChildren = $(form).closest(".comments_subtree")
+          .find(".comments_subtree").length;
+        if (isEdit && hasChildren) {
+          $(parsedHTML).find(".comment_parent_tree_line:first")
+            .removeClass("no_children");
+        }
+
+        if ($(form).find("#parent_comment_short_id").length) {
+          // reply to comment
+          $(form).closest(".comments_subtree")
+            .find(".comment_parent_tree_line:first").removeClass("no_children");
+          $(form).closest(".comment").replaceWith(parsedHTML);
+        } else {
+          // reply to story
+          $(form).parent(".comment").replaceWith(parsedHTML);
+        }
+      })
+        .always(() => done())
+        .fail(function(req, err) { alert(`could not post comment: ${err}`) })
+    })
+  },
+
+  // Prevent a button/link from working while an AJAX request is in progress;
+  // otherwise smashing a "post comment" button will post a comment twice.
+  loading: function(btn, f) {
```

Comment:
Why require the button as an argument? There's only one submit button in the form and it's very unlikely there'll be a second. Could you just select it out of the form like `$(f).querySelector('button[type="submit"]')`?

Reply:
I mostly just copied this approach from another application I wrote, where it's more useful. I guess it could be removed here, but to be honest I'd be in favour of just keeping it, as it makes it easier to re-use just in case.

For example, it could be added to e.g. the "reply" button too (I noticed that can sometimes take a second or two for the reply form to load, and a bit of feedback wouldn't hurt, and with this it should be very easy to add), or maybe the "preview" button.